diff --git a/components/EditLink.astro b/components/EditLink.astro
index 74be4c8f79d4c48b46bf92b7a2750f412a98abac..fbc2fe2b6ffc7aa51e1f398cf0e6eaf64db58f69 100644
--- a/components/EditLink.astro
+++ b/components/EditLink.astro
@@ -1,27 +1,38 @@
 ---
-import Icon from '../user-components/Icon.astro';
-import type { Props } from '../props';
+import Icon from '../user-components/Icon.astro'
+import type { Props } from '../props'
 
-const { editUrl, labels } = Astro.props;
+let { editUrl, labels } = Astro.props
+console.log({ editUrl })
+if (editUrl?.pathname.includes('/reference/')) {
+  if (editUrl.pathname.includes('/reference/index')) {
+    editUrl = undefined
+  } else {
+    editUrl = new URL(
+      editUrl.pathname.replace('/src/content/docs/reference/', '/'),
+      editUrl.origin
+    )
+  }
+}
 ---
 
 {
-	editUrl && (
-		<a href={editUrl} class="sl-flex">
-			<Icon name="pencil" size="1.2em" />
-			{labels['page.editLink']}
-		</a>
-	)
+  editUrl && (
+    <a href={editUrl} class="sl-flex">
+      <Icon name="pencil" size="1.2em" />
+      {labels['page.editLink']}
+    </a>
+  )
 }
 
 <style>
-	a {
-		gap: 0.5rem;
-		align-items: center;
-		text-decoration: none;
-		color: var(--sl-color-gray-3);
-	}
-	a:hover {
-		color: var(--sl-color-white);
-	}
+  a {
+    gap: 0.5rem;
+    align-items: center;
+    text-decoration: none;
+    color: var(--sl-color-gray-3);
+  }
+  a:hover {
+    color: var(--sl-color-white);
+  }
 </style>
